File: app/src/main/java/com/mifos/apache/fineract/data/models/loan/LoanParameters.java
package com.mifos.apache.fineract.data.models.loan;

import java.util.List;

/**
 * @author Rajan Maurya
 *         On 12/07/17.
 */

public class LoanParameters {

    private String customerIdentifier;
    private List<CreditWorthinessSnapshot> creditWorthinessSnapshots;
    private Double maximumBalance;
    private TermRange termRange;
    private PaymentCycle paymentCycle;

    public String getCustomerIdentifier() {
        return customerIdentifier;
    }

    public void setCustomerIdentifier(String customerIdentifier) {
        this.customerIdentifier = customerIdentifier;
    }

    public List<CreditWorthinessSnapshot> getCreditWorthinessSnapshots() {
        return creditWorthinessSnapshots;
    }

    public void setCreditWorthinessSnapshots(
            List<CreditWorthinessSnapshot> creditWorthinessSnapshots) {
        this.creditWorthinessSnapshots = creditWorthinessSnapshots;
    }

    public Double getMaximumBalance() {
        return maximumBalance;
    }

    public void setMaximumBalance(Double maximumBalance) {
        this.maximumBalance = maximumBalance;
    }

    public TermRange getTermRange() {
        return termRange;
    }

    public void setTermRange(TermRange termRange) {
        this.termRange = termRange;
    }

    public PaymentCycle getPaymentCycle() {
        return paymentCycle;
    }

    public void setPaymentCycle(PaymentCycle paymentCycle) {
        this.paymentCycle = paymentCycle;
    }
}

File: app/src/main/java/org/apache/fineract/data/models/loan/LoanParameters.kt
package org.apache.fineract.data.models.loan

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize
import kotlinx.android.parcel.RawValue

/**
 * @author Rajan Maurya
 * On 12/07/17.
 */

@Parcelize
data class LoanParameters(
        @SerializedName("customerIdentifier") var customerIdentifier: String? = null,
        @SerializedName("creditWorthinessSnapshots") var creditWorthinessSnapshots:
        List<CreditWorthinessSnapshot>? = null,
        @SerializedName("maximumBalance") var maximumBalance: Double? = null,
        @SerializedName("termRange") var termRange: @RawValue TermRange? = null,
        @SerializedName("paymentCycle") var paymentCycle: @RawValue PaymentCycle? = null
) : Parcelable